californium-core/src/main/java/org/eclipse/californium/core/network/BaseMatcher.java

```diff
@@ -18,30 +18,54 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import org.eclipse.californium.core.coap.MessageObserverAdapter;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.coap.Response;
+import org.eclipse.californium.core.network.Exchange.KeyToken;
+import org.eclipse.californium.core.network.Exchange.Origin;
 import org.eclipse.californium.core.network.config.NetworkConfig;
+import org.eclipse.californium.core.observe.NotificationListener;
+import org.eclipse.californium.core.observe.Observation;
+import org.eclipse.californium.core.observe.ObservationStore;
+import org.eclipse.californium.elements.CorrelationContext;
 
 /**
- * A base class for implementing Matchers that provides support for
- * using a {@code MessageExchangeStore}.
+ * A base class for implementing Matchers that provides support for using a
+ * {@code MessageExchangeStore}.
  */
 public abstract class BaseMatcher implements Matcher {
 
 	private static final Logger LOG = Logger.getLogger(BaseMatcher.class.getName());
 	protected final NetworkConfig config;
 	protected boolean running = false;
 	protected MessageExchangeStore exchangeStore;
+	protected ObservationStore observationStore;
+	private NotificationListener notificationListener;
 
 	/**
 	 * Creates a new matcher based on configuration values.
 	 * 
 	 * @param config the configuration to use.
+	 * @param notificationListener the callback to invoke for notifications
+	 *            received from peers.
+	 * @param observationStore the object to use for keeping track of
+	 *            observations created by the endpoint this matcher is part of.
+	 * @throws NullPointerException if the configuration, notification listener,
+	 *             or the observation store is {@code null}.
 	 */
-	public BaseMatcher(final NetworkConfig config) {
+	public BaseMatcher(final NetworkConfig config, final NotificationListener notificationListener,
+			final ObservationStore observationStore) {
 		if (config == null) {
 			throw new NullPointerException("Config must not be null");
+		} else if (notificationListener == null) {
+			throw new NullPointerException("NotificationListener must not be null");
```

Comment:
IIRC the UDPMatcher did not necessarily require a `NotificationListener`, did it?